quaint/src/connector/postgres/native/mod.rs

```diff
@@ -471,6 +474,46 @@ impl Queryable for PostgreSql {
         .await
     }
 
+    async fn parse_raw_query(&self, sql: &str) -> crate::Result<ParsedRawQuery> {
+        let stmt = self.fetch_cached(sql, &[]).await?;
+        let mut columns: Vec<ParsedRawColumn> = Vec::with_capacity(stmt.columns().len());
+        let mut parameters: Vec<ParsedRawParameter> = Vec::with_capacity(stmt.params().len());
+
+        async fn infer_type(this: &PostgreSql, ty: &PostgresType) -> crate::Result<(ColumnType, Option<String>)> {
+            let column_type = ColumnType::from(ty);
+
+            match ty.kind() {
+                PostgresKind::Enum => {
+                    let enum_name = this
+                        .query_raw("SELECT typname FROM pg_type WHERE oid = $1;", &[Value::int64(ty.oid())])
+                        .await?
+                        .into_single()?
+                        .at(0)
+                        .expect("could not find enum name")
+                        .to_string()
+                        .expect("enum name is not a string");
+
+                    Ok((column_type, Some(enum_name)))
+                }
```

Comment:
As discussed in the call yesterday, this is necessary to retrieve the actual name of the `enum` type.